packages/app/src/cli/services/context/identifiers-extensions.test.ts

```diff
@@ -1004,3 +1013,196 @@ describe('ensuredeployConfirmed: handle existent uuid managed extensions', () =>
+
+    // When
+    const CONFIG_A = await testAppConfigExtensions()
+
+    const ensureExtensionsIdsOptions = options([], [], {
+      includeDeployConfig: true,
+      configExtensions: [CONFIG_A],
+      developerPlatformClient,
+    })
+    const got = await deployConfirmed(ensureExtensionsIdsOptions, [], [REGISTRATION_CONFIG_A], {
+      extensionsToCreate,
+      validMatches,
+    })
+
+    // Then
+    expect(developerPlatformClient.createExtension).not.toHaveBeenCalled()
+    expect(got).toEqual({
+      extensions: {},
+      extensionIds: {'point-of-sale': 'C_A'},
+      extensionsNonUuidManaged: {'point-of-sale': 'UUID_C_A'},
+    })
+  })
+})
+
+describe('shiftRegistrationsAround', () => {
+  test('extension registrations have dynamic UID strategy should be returned in the same array as configuration registrations', () => {
+    // Given
+    const dynamicUidStrategyExtension = {
+      uuid: 'webhook-subscription-uuid',
+      id: 'webhook-subscription-id',
+      title: 'Webhook Subscription',
+      type: 'WEBHOOK_SUBSCRIPTION',
+    }
+
+    const extensionRegistrations = [REGISTRATION_B, dynamicUidStrategyExtension]
+    const configurationRegistrations = [
+      {
+        uuid: 'UUID_C_A',
+        id: 'C_A',
+        title: 'C_A',
+        type: 'POINT_OF_SALE',
+      },
+    ]
+    const specifications = [appPOSSpec, appWebhookSubscriptionSpec]
+
+    const dynamicUidStrategySpec = specifications.find(
+      (spec) => spec.identifier === dynamicUidStrategyExtension.type.toLowerCase(),
+    )
+    expect(dynamicUidStrategySpec?.experience).toEqual('configuration')
+    expect(dynamicUidStrategySpec?.uidStrategy).toEqual('dynamic')
+
+    // When
+    const {uuidUidStrategyExtensions, singleAndDynamicStrategyExtensions} = groupRegistrationByUidStrategy(
+      extensionRegistrations,
+      configurationRegistrations,
+      specifications,
+    )
+
+    // Then
+    expect(uuidUidStrategyExtensions).toEqual([REGISTRATION_B])
+    expect(singleAndDynamicStrategyExtensions).toEqual([configurationRegistrations[0], dynamicUidStrategyExtension])
+  })
+})
+
+describe('ensureNonUuidManagedExtensionsIds: for extensions managed in the TOML', () => {
+  test('creates the extension if no possible matches', async () => {
+    // Given
+    const remoteSources: RemoteSource[] = []
+    const localSources = [
+      await testSingleWebhookSubscriptionExtension(),
+      await testSingleWebhookSubscriptionExtension({topic: 'products/create'}),
+      await testSingleWebhookSubscriptionExtension({topic: 'products/update'}),
+      await testSingleWebhookSubscriptionExtension({topic: 'products/delete'}),
+    ]
+    const app = options(localSources, [], {
+      includeDeployConfig: true,
+      flags: [Flag.DeclarativeWebhooks],
+    }).app
+    const appId = 'appId'
+
+    let createExtensionCounter = 0
+    const developerPlatformClient = testDeveloperPlatformClient({
+      createExtension: async () => {
```

Comment:
I believe so, since in `testDeveloperPlatformClient`, `createExtension` is a function that resolves a Promise:

```
createExtension: (_input: ExtensionCreateVariables) => Promise.resolve(extensionCreateResponse),
```

Reply:
shouldn't it just infer that `createExtensionResult` returns a Promise? weird